sim_city.py

```diff
@@ -0,0 +1,106 @@
+import random
+import math
+
+BASE_CITY_INITIAL_TAX = 1000
+
+
+class City:
+
+    def __init__(self):
+        self.neighborhoods = []
+        self.base_city_tax = BASE_CITY_INITIAL_TAX
+
+    def how_much_money(self):
+        neighborhoods_tax = sum([neighborhood.calculate_tax() for neighborhood in self.neighborhoods])
+        return self.base_city_tax + neighborhoods_tax
+
+    def build_a_neighborhood(self, neighborhood_name):
+        print(f'Neighborhood \'{neighborhood_name}\' built')
+        new_neighborhood = Neighborhood(neighborhood_name)
+        self.neighborhoods.append(new_neighborhood)
+        self.base_city_tax += (self.base_city_tax * 0.1)
+        return 0
+
+    def build_a_house(self, neighborhood_name, family_members, size):
+        print(f'House built in \'{neighborhood_name}\' with {family_members} family members and of size {size}')
+        new_house = House(size, family_members)
+        neighborhood = self.__find_neighborhood(neighborhood_name)
+        neighborhood.houses.append(new_house)
+        return 0
+
+    def ruin_a_neighborhood(self, neighborhood_name):
+        print(f'Neighborhood \'{neighborhood_name}\' ruined')
+        neighborhood = self.__find_neighborhood(neighborhood_name)
+        self.neighborhoods.remove(neighborhood)
+        self.base_city_tax += (self.base_city_tax * 0.05)
+
+    def build_a_park(self, neighborhood_name):
+        print(f'Park built in \'{neighborhood_name}\'')
+        neighborhood = self.__find_neighborhood(neighborhood_name)
+        neighborhood.parks += 1
+
+    def __find_neighborhood(self, neighborhood_name):
+        return [n for n in self.neighborhoods if n.name == neighborhood_name][0]
```

Comment:
List comprehension is not suitable here, just use a naive for (you go over the while list when you just need to find the one with the suitable name)

Reply:
fixed